src/libsyntax/ast.rs

```diff
@@ -1441,8 +1441,17 @@ pub struct MacroDef {
     pub legacy: bool,
 }
 
-// Clippy uses Hash and PartialEq
-#[derive(Clone, RustcEncodable, RustcDecodable, Debug, Copy, Hash, PartialEq, HashStable_Generic)]
+#[derive(
```

Comment:
Please avoid changing this diff -- add a new line of `#[derive(...)]` instead and put `HashStable_Generic` in there.